lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java

```diff
@@ -1903,18 +1903,21 @@ public Object readSqlResultRecord(final ResultSet result) {
 	}
 
 	public List<SimplePage> getTopLevelPages(final String siteId) {
-		DetachedCriteria d = DetachedCriteria.forClass(SimplePage.class).add(Restrictions.eq("siteId", siteId))
-			.add(Restrictions.disjunction()
-				.add(Restrictions.isNull("owner"))
-				.add(Restrictions.eq("owned", true)))
-			.add(Restrictions.isNull("parent"));
+	    // set of all top level pages, actually the items pointing to them                                                                       
+	    List<SimplePageItem> sitePages =  findItemsInSite(toolManager.getCurrentPlacement().getContext());
```

Comment:
Why are you using toolManager.getCurrentPlacement().getContext() when siteId is passed to this function?